neutrino.go

```diff
@@ -477,6 +476,62 @@ type Config struct {
 	NameResolver func(host string) ([]net.IP, error)
 }
 
+// DataDir is a functional option that allows callers of NewChainService to set the
+// data directory for storing header information.
+func DataDir(dataDir string) func(*Config) {
+	return func(c *Config) {
+		c.DataDir = dataDir
```

Comment:
When adding functional options, I think we should unexport the set fields?

Reply:
Good call. It's better to keep scope as small as possible. I'll change the fields to private